core/src/main/java/edu/wpi/grip/core/operations/composite/ContoursReport.java

```diff
@@ -21,27 +36,81 @@ public ContoursReport(MatVector contours, int rows, int cols) {
         this.cols = cols;
     }
 
-    public void setContours(MatVector contours) {
-        this.contours = contours;
+    public int getRows() {
+        return this.rows;
+    }
+
+    public int getCols() {
+        return this.cols;
     }
 
     public MatVector getContours() {
         return this.contours;
     }
 
-    public void setRows(int rows) {
-        this.rows = rows;
+    /**
+     * Compute the bounding boxes of all contours (if they haven't already been computed).  Bounding boxes are used
+     * to compute several different properties, so it's probably not a good idea to compute them over and over again.
+     */
+    private synchronized List<Rect> computeBoundingBoxes() {
+        if (!boundingBoxes.isPresent()) {
+            List<Rect> bb = new ArrayList<>();
```

Comment:
This is going to be run at 30 frame per second. We don't need to make the garbage collector deal with over allocated space.